pyqtorch/core/operation.py

```diff
@@ -262,12 +301,152 @@ def CNOT(state: torch.Tensor, qubits: ArrayLike, N_qubits: int) -> torch.Tensor:
     Returns:
         torch.Tensor: the resulting state after applying the gate
     """
+
     if ops_cache.enabled:
         store_operation("CNOT", qubits)
+    
+    return ControlledOperationGate(state, qubits, N_qubits, NOTMAT)
+
+
+def CRX(theta: torch.Tensor, state: torch.Tensor, qubits: ArrayLike, N_qubits: int) -> torch.Tensor:
+    """Controlled RX rotation gate with two-qubits support
+
+    Args:
+        state (torch.Tensor): the input quantum state, of shape `(N_0, N_1,..., N_N, batch_size)`
+        qubits (ArrayLike): list of qubit indices where the gate will operate
+        N_qubits (int): the number of qubits in the system
+        theta (torch.Tensor): 1D-tensor holding the values of the parameter
+
+    Returns:
+        torch.Tensor: the resulting state after applying the gate
+    """
+    if ops_cache.enabled:
+        store_operation("CRX", qubits, param=theta)
+
+    operation_matrix: torch.Tensor = get_parametrized_matrix_for_operation("RX", theta)
+    return ControlledOperationGate(state, qubits, N_qubits, operation_matrix)
+
+
+def CRY(theta: torch.Tensor, state: torch.Tensor, qubits: ArrayLike, N_qubits: int) -> torch.Tensor:
+    """Controlled RY rotation gate with two-qubits support
+
+    Args:
+        state (torch.Tensor): the input quantum state, of shape `(N_0, N_1,..., N_N, batch_size)`
+        qubits (ArrayLike): list of qubit indices where the gate will operate
+        N_qubits (int): the number of qubits in the system
+        theta (torch.Tensor): 1D-tensor holding the values of the parameter
+
+    Returns:
+        torch.Tensor: the resulting state after applying the gate
+    """
+
+    if ops_cache.enabled:
+        store_operation("CRY", qubits, param=theta)
+
+    operation_matrix: torch.Tensor = get_parametrized_matrix_for_operation("RY", theta)
+    return ControlledOperationGate(state, qubits, N_qubits, operation_matrix)
+
+
+def CRZ(theta: torch.Tensor, state: torch.Tensor, qubits: ArrayLike, N_qubits: int) -> torch.Tensor:
+    """Controlled RZ rotation gate with two-qubits support
+
+    Args:
+        state (torch.Tensor): the input quantum state, of shape `(N_0, N_1,..., N_N, batch_size)`
+        qubits (ArrayLike): list of qubit indices where the gate will operate
+        N_qubits (int): the number of qubits in the system
+        theta (torch.Tensor): 1D-tensor holding the values of the parameter
+
+    Returns:
+        torch.Tensor: the resulting state after applying the gate
+    """
+
+    if ops_cache.enabled:
+        store_operation("CRZ", qubits, param=theta)
```

Comment:
@dominikandreasseitz Make sure to update also the mapping in the `converters/to_qiskit.py` module to make sure that Qiskit picks up the right gates when performing the conversion.

Reply:
todo for tomorrow (7.2)

Reply:
done